canopen/objectdictionary/eds.py

```diff
@@ -122,7 +122,13 @@ def build_variable(eds, section, index, subindex=0):
             pass
     if eds.has_option(section, "DefaultValue"):
         try:
-            var.default = int(eds.get(section, "DefaultValue"), 0)
+            default_value = eds.get(section, "DefaultValue")
+            #COB-ID can have a suffix of '$NODEID+' so replace this with node_id before converting
+            if '$NODEID+' in default_value:
+               var.default = int(default_value.replace('$NODEID+',''), 0) + node_id
```

Comment:
Try something like this:

```python
if '$NODEID' in default_value and node_id is not None:
    var.default = eval(default_value.replace('$NODEID', str(node_id)))
```

Then you can revert node.py.

Reply:
Good idea and that would be a good solution.

But the problem is that when you set node ID to 0 and it doesn't locate a Node ID in the EDS file, python stops on the line below my change in node.py:
`self.sdo = SdoClient(0x600 + self.id, 0x580 + self.id, object_dictionary)`
That you cannot do '+' on an integer and a 'None' type.

And since you have an actual travis test for this specific case I thought this was expected behavior.
Otherwise there is no failure when the system cannot find a node ID in the EDS file.

But, I honestly think what my use case is a bit special so I will revert the change to node.py.
The suggestion with `and node_id is not None:` should not be needed since python will do a raise before this issue if node_id is None.

Reply:
It seems `and node_id is not None:` was needed :)